### ISSUE TITLE
With React 91 typings: Cannot find namespace 'JSX'.

### ISSUE THREAD

**Issue by Philipp91** [clauderic/dnd-kit#1559]
I previously posted about this [here](https://github.com/clauderic/dnd-kit/issues/1511#issuecomment-2523857814), but I don't want to spam that thread.

```
node_modules/@dnd-kit/core/dist/components/Accessibility/Accessibility.d.ts:9:129 - error TS2503: Cannot find namespace 'JSX'.

9 export declare function Accessibility({ announcements, container, hiddenTextDescribedById, screenReaderInstructions, }: Props): JSX.Element | null;
```

We basically need `s/JSX.Element/React.JSX.Element` to fix this. There are two occurrences in this repo itself, for which I will send a PR. Though this requires updating to v18 typings first.

But most of the `JSX.Element` occurrences seem to come from the build process, which materializes types that were previously not spelled out explicitly.

**Comment by Philipp91** [comment]
The `dist/` directories aren't committed to this Git repository. But my current understanding is that they _are_ part of the NPM package, so they're not built on the library user's machine, but on the library maintainer's machine when building the release.

When I clone the repo and run `yarn install`, I get the correct typings already:

```
$ yarn install
$ cat packages/core/dist/components/Accessibility/Accessibility.d.ts
...
export declare function Accessibility({ announcements, container, hiddenTextDescribedById, screenReaderInstructions, }: Props): React.JSX.Element | null;
```

Whereas in my main app's repo from NPM I got this installed:

```
export declare function Accessibility({ announcements, container, hiddenTextDescribedById, screenReaderInstructions, }: Props): JSX.Element | null;
```

and I see the same on https://www.npmjs.com/package/@dnd-kit/core/v/6.3.1?activeTab=code.

Given that the `yarn.lock` should fix all the versions, it's not clear to me why I get a different result than what's released. 

@clauderic I assume it's GitHub Actions like [this one](https://github.com/clauderic/dnd-kit/actions/runs/12184596050/job/33988855568) that build the releases. Just as another data point, on your own machine, do you get `React.JSX.Element` or `JSX.Element`? Then we can compare how the environments differ to figure out how we can get the release environment to produce the new-style typings.

**Comment by Philipp91** [comment]
@clauderic I know you're working on the new `@dnd-kit/react` refactoring. While I don't know when that will be done, even if we assume that's next week already, it's probably still not realistic to get _everyone_ to migrate to that before they migrate to React v19.  So it seems important to make the legacy library compatible with React v19 too.

**Comment by clauderic** [comment]
This issue was opened against a previous version of dnd-kit. The library has since undergone a major rewrite.

It has been rebuilt from the ground up with support for **multiple frameworks** (React, Vue, Svelte, Solid, and vanilla JavaScript), improved performance, and a new API.

- **Documentation**: https://dndkit.com
- **Migration guide**: https://dndkit.com/react/migration
- **GitHub**: https://github.com/clauderic/dnd-kit

If this issue is still relevant to the latest version, please open a new issue.
